plugin/allow/allow.go

```diff
@@ -0,0 +1,40 @@
+package allow
+
+import (
+    "net"
+
+    "github.com/coredns/coredns/plugin"
+    "github.com/coredns/coredns/request"
+
+    "github.com/miekg/dns"
+    "golang.org/x/net/context"
+)
+
+type Allow struct {
+    Next     plugin.Handler
+    Cidrs  []string
+}
+
+func (rw Allow) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+    state := request.Request{W: w, Req: r}
+    ip := state.IP()
+
+    for _, it := range rw.Cidrs {
+        _, cidr, err := net.ParseCIDR(it)
+        if err != nil {
+            return 0, err
+        }
+        if cidr.Contains(net.ParseIP(ip)) {
```

Comment:
Lines 23-26 can be done during config parse instead of inline with request handling.